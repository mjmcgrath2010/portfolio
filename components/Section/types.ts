import { ReactElement } from "react";

interface SectionProps {
  children: ReactElement | ReactElement[] | string;
  className?: string;
  color?: "light" | "dark" | "accent";
}

export default SectionProps;
