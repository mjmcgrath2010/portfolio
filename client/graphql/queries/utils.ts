/*
 * Example usage:
 *   export const getStaticProps = () => mapQueriesToProps([{ posts: getPosts }]);
 */
type QueryProps = {
  props: Object;
};

export const mapQueriesToProps = async (queries: any) =>
  await (async () => {
    const data = await queries.reduce(
      async (acc: QueryProps, queryObj: Function) => {
        const [prop, fn] = Object.entries(queryObj)[0];
        try {
          const { props } = acc;
          const data = await fn();
          return {
            props: {
              ...props,
              [prop]: data,
            },
          };
        } catch (e) {
          return {
            props: {
              error: "Oops, something went wrong",
            },
          };
        }
      },
      { props: {} }
    );
    return data;
  })();
