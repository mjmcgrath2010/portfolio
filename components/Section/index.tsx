import React from "react";
import styled from "styled-components";
import SectionProps from "./types";

const Container = styled.div<SectionProps>`
  background: ${({ theme, color = "light" }) =>
    theme.palette.background[color]};
  min-height: 300px;
  overflow: hidden;
  display: block;
  position: relative;
  padding: ${({ theme }) => theme.spacing.xl};
`;

const ContentContainer = styled.div`
  max-width: 1440px;
  margin: 0 auto;
`;

const Section = ({ children, className, color }: SectionProps) => (
  <Container className={className} color={color}>
    <ContentContainer>{children}</ContentContainer>
  </Container>
);

Section.defaultProps = {
  color: "light",
  className: "",
};

export default Section;
