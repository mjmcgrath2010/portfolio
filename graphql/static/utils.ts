export const mapQueriesToProps = async (queries: any) =>
  await (async () => {
    const data = await queries.reduce(
      async (acc, queryObj) => {
        const [prop, fn] = Object.entries(queryObj)[0];
        try {
          const { props } = acc;
          const data = await fn();

          return {
            props: {
              ...props,
              [prop]: data,
            },
          };
        } catch (e) {
          return {
            error: e.message,
          };
        }
      },
      { props: {} }
    );
    return data;
  })();
