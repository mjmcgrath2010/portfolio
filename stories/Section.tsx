/* eslint-disable import/no-anonymous-default-export */
import React from "react";
import Section from "../components/Section";
import SectionProps from "../components/Section/types";

export default {
  title: "Components/Section",
  component: Section,
  argTypes: {},
  parameters: {},
};

export const DefaultStory = ({ ...rest }: SectionProps) => (
  <Section {...rest} />
);

DefaultStory.args = {
  children: "Section!",
};

DefaultStory.storyName = "Section - Default Story";
