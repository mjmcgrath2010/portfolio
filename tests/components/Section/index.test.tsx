import React from "react";
import Section from "@components/Section";

import { render, screen } from "@test/utils";

describe("<Section />", () => {
  it("matches the snapshot", () => {
    const view = render(<Section>hi</Section>);

    expect(view).toMatchSnapshot();
  });
});
