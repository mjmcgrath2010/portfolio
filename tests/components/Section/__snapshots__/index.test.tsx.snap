// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  background: rgb(223,223,223);
  min-height: 300px;
  overflow: hidden;
  display: block;
  position: relative;
  padding: 32px;
}

.c1 {
  max-width: 1440px;
  margin: 0 auto;
}

<body>
    <div>
      <div
        class="c0"
        color="light"
      >
        <div
          class="c1"
        >
          hi
        </div>
      </div>
    </div>
  </body>,
  "container": .c0 {
  background: rgb(223,223,223);
  min-height: 300px;
  overflow: hidden;
  display: block;
  position: relative;
  padding: 32px;
}

.c1 {
  max-width: 1440px;
  margin: 0 auto;
}

<div>
    <div
      class="c0"
      color="light"
    >
      <div
        class="c1"
      >
        hi
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
